src/os/systemd.zig

```diff
@@ -0,0 +1,64 @@
+const std = @import("std");
+const builtin = @import("builtin");
+
+const log = std.log.scoped(.systemd);
+
+/// Returns true if the program was launched as a systemd service.
+///
+/// On Linux, this returns true if the program was launched as a systemd
+/// service. It will return false if Ghostty was launched any other way.
+///
+/// For other platforms and app runtimes, this returns false.
+pub fn launchedBySystemd() bool {
+    return switch (builtin.os.tag) {
+        .linux => linux: {
+            // On Linux, systemd sets the `INVOCATION_ID` (v232+) and the
+            // `JOURNAL_STREAM` (v231+) environment variables. If these
+            // environment variables are not present we were not launched by
+            // systemd.
+
+            if (std.posix.getenv("INVOCATION_ID") == null) break :linux false;
+            if (std.posix.getenv("JOURNAL_STREAM") == null) break :linux false;
+
+            // If `INVOCATION_ID` and `JOURNAL_STREAM` are present, check to make sure
+            // that our parent process is actually `systemd`, not some other terminal
+            // emulator that doesn't clean up those environment variables.
+
+            const ppid = std.os.linux.getppid();
+
+            // If the parent PID is 1 we'll assume that it's `systemd` as other init systems
+            // are unlikely.
+
+            if (ppid == 1) break :linux true;
```

Comment:
This seems like quite an assumption. This is true _today_ because most distros use systemd, but historically any init system would launch children from pid 1 no?

Reply:
Nevermind, I see the guards before this that verify systemd via env vars.